encoding/ccf/cache.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2022-2023 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ccf
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/onflow/cadence"
+)
+
+// cachedSortedFieldIndex contains sorted field index of Cadence composite types.
+var cachedSortedFieldIndex sync.Map // key: cadence.CompositeType, value: []int
```

Comment:
@turbolent `cachedSortedFieldIndex` is global so it can be reused across multiple CCF encodings.  However, `cadence.CompositeType` may not be the best choice as map key.

Do you think it is useful to have a global cache of sorted composite field indexes across multiple CCF encodings?  If so, does composite type (with the same Cadence type ID) always have the same unsorted field order?

Reply:
Moved `cachedSortedFieldIndex` from global into `Encoder` with commit cd55481beedde1b8da789e485fad97adc8dd1f88.  

This change causes extra memory allocation.  I can optimize this in later PR.

Reply:
Nice! We can potentially reuse the information across multiple executions as an internal optimization, it's just nice to have the library be stateless 👍